### PR TITLE
Cleaning

### PR DESCRIPTION
**Pull request by Gabriel-Darbord**

### CODE REVIEW

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 1]
```diff
@@ -4163,20 +4183,14 @@ StackInterpreter >> checkOkayStackPage: thePage [
...
 
-	0 to: numStackPages - 1 do:
-		[:i|
-		thePage := stackPages stackPageAt: i.
-		(stackPages isFree: thePage) ifFalse:
-			[ok := ok and: [ (self checkOkayStackPage: thePage)]] ].
+	self usedStackPagesDo: [ :thePage |
+		(self checkOkayStackPage: thePage) ifFalse: [ ^ false ] ].
 
-	^ok
```
Comment (Gabriel-Darbord) [inline smalltalksrc/VMMaker/StackInterpreter.class.st:4179]: Maybe this was printing to the terminal so we should still iterate all pages...

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 2]
```diff
@@ -2188,6 +2198,20 @@ StackInterpreter >> assertValidStackLimits: ln [
 		l: ln
 ]
 
+{ #category : 'object memory support' }
+StackInterpreter >> baseFrameArguments: baseFrameFP inPage: thePage do: aBlock [
+	"In base frames the caller saved ip field holds the caller context."
```
Comment (Gabriel-Darbord) [inline smalltalksrc/VMMaker/StackInterpreter.class.st]: We lost the good comment here!

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 3]
```diff
@@ -6877,6 +6887,28 @@ StackInterpreter >> frameStackedReceiverOffsetNumArgs: numArgs [
...
+	| theSP theFP callerFP |
+	theSP := thePage headSP.
+	theFP := thePage headFP.
+
+	"Skip the instruction pointer on top of stack of inactive pages."
+	thePage = stackPage ifFalse: [
+		theSP := theSP + objectMemory wordSize ].
+
+	[
+	aBlock value: theFP value: theSP.
```
Comment (Gabriel-Darbord) [inline smalltalksrc/VMMaker/StackInterpreter.class.st:6909]: We should probably put the instruction pointer here too

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 4]
```diff
@@ -6877,6 +6887,28 @@ StackInterpreter >> frameStackedReceiverOffsetNumArgs: numArgs [
 	^FoxCallerSavedIP + objectMemory wordSize + (numArgs << objectMemory shiftForWord)
 ]
 
+{ #category : 'object memory support' }
+StackInterpreter >> framesInPage: thePage do: aBlock [
+
+	| theSP theFP callerFP |
+	theSP := thePage headSP.
+	theFP := thePage headFP.
```
Comment (Gabriel-Darbord) [inline smalltalksrc/VMMaker/StackInterpreter.class.st:6902]: Warning: "conflicting types char * sqInt for theFP in framesInPage:do:"

**Review by guillep — APPROVED** [review]

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 5]
```diff
@@ -4188,14 +4188,15 @@ StackInterpreter >> checkOkayStackPage: thePage [
...
 	writeBack ifTrue: [ self writeBackHeadFramePointers ].
-
+	
+	ok := true.
 	self usedStackPagesDo: [ :thePage |
-		(self checkOkayStackPage: thePage) ifFalse: [ ^ false ] ].
-
-	^ true
+		ok := ok and: [ (self checkOkayStackPage: thePage) ] ].
+	^ ok
```
Comment (Gabriel-Darbord) [inline smalltalksrc/VMMaker/StackInterpreter.class.st:4199]: Can't we leave the early return because it uses a lazy `and:` anyway?

Comment (guillep) [reply smalltalksrc/VMMaker/StackInterpreter.class.st:4199]: the thing is that the checkOkStackPage: and child have side effects that print things...

Comment (Gabriel-Darbord) [reply smalltalksrc/VMMaker/StackInterpreter.class.st:4199]: Sure but it's using `and:`, so it stop calling `checkOkayStackPage:` after the first time it returns false, or am I missing something?

**Pull request merged**